chainercv/datasets/flic/flic_keypoint_dataset.py

```diff
@@ -0,0 +1,137 @@
+import glob
+import os
+
+import numpy as np
+
+import chainer
+from chainercv.datasets.flic import flic_utils
+from chainercv import utils
+
+try:
+    from scipy.io import loadmat
+    _scipy_available = True
+except (ImportError, TypeError):
+    _scipy_available = False
+
+
+class FLICKeypointDataset(chainer.dataset.DatasetMixin):
+
+    """`Frames Labaled in Cinema (FLIC)`_ dataset  with annotated keypoints.
+
+    .. _`Frames Labaled in Cinema (FLIC)`:
+        https://bensapp.github.io/flic-dataset.html
+
+    An index corresponds to each image.
+
+    When queried by an index, this dataset returns the corresponding
+    :obj:`img, keypoint`, which is a tuple of an image and keypoints
+    that indicates visible keypoints in the image.
+    The data type of the two elements are :obj:`float32, float32`.
+
+    The keypoints are packed into a two dimensional array of shape
+    :math:`(K, 2)`, where :math:`K` is the number of keypoints.
+    Note that :math:`K=29` in FLIC dataset. Also note that not all
+    keypoints are visible in an image. When a keypoint is not visible,
+    the values stored for that keypoint are :obj:`~numpy.nan`. The second axis
+    corresponds to the :math:`y` and :math:`x` coordinates of the
+    keypoints in the image.
+
+    The torso bounding box is a one-dimensional array of shape :math:`(4,)`.
+    The elements of the bounding box corresponds to
+    :math:`(y_{min}, x_{min}, y_{max}, x_{max})`, where the four attributes are
+    coordinates of the top left and the bottom right vertices.
+    This information can optionally be retrieved from the dataset
+    by setting :obj:`return_torsobox = True`.
+
+    Args:
+        data_dir (string): Path to the root of the training data. If this is
+            :obj:`auto`, this class will automatically download data for you
+            under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/cub`.
+        split ({'train', 'test'}): Select from dataset splits used in
+            the FLIC dataset.
+        return_torsobox (bool): If :obj:`True`, this returns a bounding box
+            around the torso. The default value is :obj:`False`.
+        skip_bad (bool): If :obj:`True`, the data which have :obj:`isbad = 1`
```

Comment:
How about `use_bad` and `use_unchecked`?
This style is consistent with the other datasets.
https://github.com/chainer/chainercv/blob/master/chainercv/datasets/voc/voc_bbox_dataset.py#L70